src/vellum/lib/test_suite.py

```diff
@@ -0,0 +1,175 @@
+
+    def _get_test_suite_id_by_name(self, name: str) -> str:
+        # TODO: We need to EITHER:
+        # - Allow for all of these methods to be called with `name`
+        # - OR, we need to expose a get test suite id by name method and save the `id`
+        return uuid4()
+
+    def run_external(self, external_evaluator: Callable[..., dict[str, ValidOutput]]) -> VellumTestSuiteResults:
+        test_cases = self.client.test_suites.list_test_suite_test_cases(id=self.test_suite_id)
+        executions: list[ExternalTestCaseExecution] = []
+
+        # TODO - we need either:
+        # - Retrieve input_variables[i].key from test_suites
+        # - OR add a `name` field to `TestCaseVariableValue`
+        input_variable_name_by_id = {}
+
+        for test_case in test_cases.results:
+            inputs = test_case.input_values
+            external_input = {input_variable_name_by_id.get(inp.variable_id): inp.value for inp in inputs}
+            external_outputs = external_evaluator(
+                **external_input,
+            )
+
+            outputs = []
+            for external_output_key, external_output_value in external_outputs.items():
+                if isinstance(external_output_value, str):
+                    # Doesn't currently work - requires NamedTestCaseVariableValueRequest_String
+                    # Same for all other options below
+                    outputs.append(
+                        NamedTestCaseStringVariableValueRequest(
+                            value=external_output_value,
+                            name=external_output_key,
+                        )
+                    )
+
+                elif isinstance(external_output_value, float):
+                    outputs.append(
+                        NamedTestCaseNumberVariableValueRequest(
+                            value=external_output_value,
+                            name=external_output_key,
+                        )
+                    )
+
+                elif isinstance(external_output_value, dict):
+                    outputs.append(
+                        NamedTestCaseJsonVariableValueRequest(
+                            value=external_output_value,
+                            name=external_output_key,
+                        )
+                    )
+
+                # TODO: Handle the rest of the types
+
+            executions.append(
+                ExternalTestCaseExecution(
+                    test_case_id=test_case.id,
+                    outputs=outputs,
+                )
+            )
+
+        test_suite_run = self.client.test_suite_runs.create(
+            test_suite_id=self.test_suite_id,
+            # TODO: Also doesn't work yet for the same reason as variables above
+            exec_config=TestSuiteRunExternalExecConfigRequest(
+                data=TestSuiteRunExternalExecConfigDataRequest(
+                    executions=executions,
+                ),
+            ),
```

Comment:
blocking 5 - no decision here, I just forgot to re-expose `TestSuiteRunExternalExecConfigRequest` in our api client.

I do want to take this opportunity to demonstrate why I lean away from `type/data` and more for `type/[...attrs]` - the consumer needs to import two dataclasses that say basically the same thing for an unclear value add